segments/weather.sh

```diff
@@ -118,8 +114,14 @@ __yrno() {
 		fi
 		# condition_symbol=$(__get_yrno_condition_symbol "$condition" "$sunrise" "$sunset")
 		condition_symbol=$(__get_yrno_condition_symbol "$condition")
-		echo "${condition_symbol} ${degree}°$(echo "$TMUX_POWERLINE_SEG_WEATHER_UNIT" | tr '[:lower:]' '[:upper:]')" | tee "${tmp_file}"
+		# Write the <content @ date>, separated by 2 spaces and @, so we can fetch it later on without having to call 'stat'
+		echo "${condition_symbol} ${degree}°$(echo "$TMUX_POWERLINE_SEG_WEATHER_UNIT" | tr '[:lower:]' '[:upper:]')@$(date +%s)" > "$cache_weather_file"
+		__read_file_content "$cache_weather_file"
+		return
 	fi
+
+	__read_file_content "$cache_weather_file"
+	return
```

Comment:
aaaand  just return some error value I guess here. because if we get here it means we could not retrieve the weather for some reason or the other.

```suggestion
	return 1
```